### PR TITLE
[IMPROVEMENT] Added the Ubooquity bookmark feature and more fixes to OPDS

### PR DESCRIPTION
**Pull request by jmcrfp**
## Status
**READY**

## Migrations
YES 

## Description
Added the Ubooquity bookmark feature and more fixes to OPDS

### CODE REVIEW

File: comixed-rest-api/src/main/java/org/comixed/controller/opds/OPDSAcquisitionFeed.java [hunk 1]
```diff
@@ -44,7 +41,8 @@
 
     private String title;
 
-    private ZonedDateTime updated;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss'Z'")
+    private Date updated;
```
Comment (mcpierce) [inline comixed-rest-api/src/main/java/org/comixed/controller/opds/OPDSAcquisitionFeed.java]: I see this is changing the implementation from using ZonedDateTime to Date. Is there a technical benefit for that change?

File: comixed-rest-api/src/main/java/org/comixed/controller/ubooquity/UbooquityController.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * <code>UbooquityController</code> allows the remote agent request reading bookmark
+ * and set reading bookmark of a book
+ *
+ * @author João França
+ * @author Darryl L. Pierce
+ */
+@RestController
+@RequestMapping("/user-api")
+public class UbooquityController
```
Comment (mcpierce) [inline comixed-rest-api/src/main/java/org/comixed/controller/ubooquity/UbooquityController.java]: I'm not sure I want to have classes in CX that are named after other applications like Ubooquity. If this class is acting as a bookmarks endpoint, then let's give it a different name that reflects what it's doing in that context, someting like "OPDSBookmarkController

File: comixed-rest-api/src/main/java/org/comixed/model/ubooquity/UbooquityBookmark.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+package org.comixed.model.ubooquity;
+
+/**
+ * <code>OPDSAuthor</code> provides authors of a OPDS entry.
+ *
+ * @author João França
+ * @author Darryl L. Pierce
+ *
+ */
+public class UbooquityBookmark
```
Comment (mcpierce) [inline comixed-rest-api/src/main/java/org/comixed/model/ubooquity/UbooquityBookmark.java]: Like with the controller, I'd prefer we remove the word "Ubooquity" from this class and just call it "Bookmark".

File: comixed-rest-api/src/main/java/org/comixed/controller/opds/OPDSAcquisitionFeed.java (hunk 1 above)
Comment (jmcrfp) [reply comixed-rest-api/src/main/java/org/comixed/controller/opds/OPDSAcquisitionFeed.java]: The Kuboo app was having trouble reading in that format and to make it in the same format of the OPDS spec https://en.wikipedia.org/wiki/OPDS#Example_of_OPDS_1.1_content

File: comixed-rest-api/src/main/java/org/comixed/controller/ubooquity/UbooquityController.java (hunk 2 above)
Comment (jmcrfp) [reply comixed-rest-api/src/main/java/org/comixed/controller/ubooquity/UbooquityController.java]: I made the change.

File: comixed-rest-api/src/main/java/org/comixed/model/ubooquity/UbooquityBookmark.java (hunk 3 above)
Comment (jmcrfp) [reply comixed-rest-api/src/main/java/org/comixed/model/ubooquity/UbooquityBookmark.java]: I made the change to OPDSBookmark

**Pull request merged**